packages/next-swc/crates/next-core/src/util.rs

```diff
@@ -347,14 +352,202 @@ fn parse_config_from_js_value(module: Vc<Box<dyn Module>>, value: &JsValue) -> N
     config
 }
 
-#[turbo_tasks::function]
+/// Loads a next.js template, replaces `replacements` and `injections` and makes
+/// sure there are none left over.
+// TODO: should this be a turbo tasks function?
+// #[turbo_tasks::function]
 pub async fn load_next_js_template(
+    path: &str,
     project_path: Vc<FileSystemPath>,
-    file: String,
-) -> Result<Vc<Rope>> {
-    let file_path = virtual_next_js_template_path(project_path, file);
+    replacements: IndexMap<&'static str, String>,
+    injections: IndexMap<&'static str, String>,
+) -> Result<Vc<Box<dyn Source>>> {
+    let path = virtual_next_js_template_path(project_path, path.to_string());
+
+    let content = &*file_content_rope(path.read()).await?;
+    let content = content.to_str()?.to_string();
+
+    let parent_path = path.parent();
+    let parent_path_value = &*parent_path.await?;
+
+    let package_root = get_next_package(project_path).parent();
+    let package_root_value = &*package_root.await?;
+
+    /// See [regex::Regex::replace_all].
+    fn replace_all<E>(
+        re: &regex::Regex,
+        haystack: &str,
+        mut replacement: impl FnMut(&regex::Captures) -> Result<String, E>,
+    ) -> Result<String, E> {
+        let mut new = String::with_capacity(haystack.len());
+        let mut last_match = 0;
+        for caps in re.captures_iter(haystack) {
+            let m = caps.get(0).unwrap();
+            new.push_str(&haystack[last_match..m.start()]);
+            new.push_str(&replacement(&caps)?);
+            last_match = m.end();
+        }
+        new.push_str(&haystack[last_match..]);
+        Ok(new)
+    }
 
-    let content = &*file_path.read().await?;
+    // Update the relative imports to be absolute. This will update any relative
+    // imports to be relative to the root of the `next` package.
+    let regex = lazy_regex::regex!("(?:from \"(\\..*)\"|import \"(\\..*)\")");
+
+    let mut count = 0;
+    let mut content = replace_all(regex, &content, |caps| {
+        let from_request = caps.get(1usize).map_or("", |c| c.as_str());
+        let import_request = caps.get(2usize).map_or("", |c| c.as_str());
```

Comment:
Do these need the `usize` annotations?
```suggestion
        let from_request = caps.get(1).map_or("", |c| c.as_str());
        let import_request = caps.get(2).map_or("", |c| c.as_str());
```

Reply:
not sure how those ended up there